docs/examples/templated-config.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: quarks.cloudfoundry.org/v1alpha1
+kind: QuarksSecret
+metadata:
+  name: generate-password-for-template
+spec:
+  type: password
+  secretName: gen-secret-for-template
+---
+apiVersion: quarks.cloudfoundry.org/v1alpha1
+kind: QuarksSecret
+metadata:
+  name: templated-config-test
+spec:
+  # Define the QuarksSecret type
+  type: templatedconfig
+  # Define the templating type, in this case "helm" (it's also the only type support as for now)
+  templateType: helm
+  # The new secret name which will have the user-defined structure
+  secretName: templated-secret
+  # Templating with helm engine.
+  # Supplied values are available under .Values as usual
+  template: 
+    foo: "{{.Values.Bar}}"
+  # The values for our template in a key, value format.
+  templateValues:
```

Comment:
three values starting with template, I think this doesn't have to be flat and should be nested below `template`


The 'values' seem to be secret (field) references.

Reply:
something like the following would be ok?  ```
template:
   secrets:
   - name: ""
      field: ""
   type: "helm"
```

Reply:
Apparentyl we store all these things below 'request' for the other types.